File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand):
    transcriptionDict = {}
    transcriptionDict['G'] = 'C'
    transcriptionDict['C'] = 'G'
    transcriptionDict['T'] = 'A'
    transcriptionDict['A'] = 'U'

    transcription = ""
    for c in dna_strand:
        if c in transcriptionDict:
            transcription += transcriptionDict[c]
        else:
            raise ValueError('error')

    return transcription

File: typescript/rna-transcription/rna-transcription.ts
/*
  G -> C
  C -> G
  T -> A
  A -> U
*/

const DNA = 'GCTA'.split('')
const RNA = 'CGAU'.split('')

function translateLetter(letter: string): string {
  const indexOf = DNA.indexOf(letter)

  if (indexOf === -1) {
    throw new Error('Invalid input DNA.')
  }

  return RNA[indexOf]
}

class Transcriptor {
  toRna(rnaString: string) {
    const asArray = [...rnaString]
    let transcribed: string = ''

    asArray.forEach((letter) => {
      transcribed += translateLetter(letter)
    })

    return transcribed
  }
}

export default Transcriptor
